xds/internal/balancer/clusterimpl/balancer_test.go

```diff
@@ -210,23 +207,25 @@ func (s) TestDropByCategory(t *testing.T) {
 		t.Fatalf("unexpected error from UpdateClientConnState: %v", err)
 	}
 
-	p2 := <-cc.NewPickerCh
-	for i := 0; i < rpcCount; i++ {
-		gotSCSt, err := p2.Pick(balancer.PickInfo{})
-		// Even RPCs are dropped.
-		if i%4 == 0 {
-			if err == nil || !strings.Contains(err.Error(), "dropped") {
-				t.Fatalf("pick.Pick, got %v, %v, want error RPC dropped", gotSCSt, err)
+	cc.WaitForPicker(ctx, func(p balancer.Picker) error {
```

Comment:
I know what you changed it from would just block until test timeout, but shoudl we do something with the return value of lastErr here? Here and elsewhere in the test.

Reply:
I had a few other comments about this, I think in summary: some of the waitForX calls you changed to are wrapped in if err := waitForX(); err != nil { t.Fatal(err) } and some aren't. I feel like all of them should be wrapped in if err..., as the picker is still expected to come. IIUC, this PR only adds the possibility of more picker updates, not removing any picker updates.